### ISSUE TITLE
Plugin generates invalid class so build fails

### ISSUE THREAD

**Issue by MichalKlusak** [yshrsmz/BuildKonfig#78]
Hi. During first build, plugin generates expect object BuildKonfig (commonMain) and actual object BuildKonfig (androidMain+jvmMain) and everything works fine. But during second build (without any changes) it regenerates expect object in commonMain to actual object so build fails due to missing expect object and redeclaration of actual objects.
Below is part of my build.gradle file.

```
apply plugin: 'org.jetbrains.kotlin.multiplatform'
apply plugin: 'com.android.library'
apply plugin: 'com.codingfeline.buildkonfig'

kotlin {
        android {
            compilations.all {
                kotlinOptions.jvmTarget = rootProject.ext.javaVersion
            }
        }
        jvm()

        sourceSets {
            all {
                languageSettings {
                    optIn("kotlinx.coroutines.ExperimentalCoroutinesApi")
                    optIn("kotlin.RequiresOptIn")
                }
            }

            commonMain {}
            jvmMain {}
            androidMain {}
            androidRc {}
            androidRelease {}
        }
}

buildkonfig {
        packageName = project.PACKAGE

        defaultConfigs {
              buildConfigField "BOOLEAN", "ABC", "true", const: true
        }

        targetConfigs {
            android {
                buildConfigField "BOOLEAN", "ABC", "true", const: true
            }

            jvm {
                buildConfigField "BOOLEAN", "ABC", "false", const: true
            }
        }
}
```

**Comment by yshrsmz** [comment]
@MichalKlusak Which Kotlin version are you using?
Also, simple repro is greatly appreciated

**Comment by Tihon-Ustinov** [comment]
> 

I use gradle-7.4.2
![image](https://user-images.githubusercontent.com/15249713/222380318-14dbdf27-259c-49dc-b043-80626baf0aae.png)

And quite simple intellij idea generated project 
![image](https://user-images.githubusercontent.com/15249713/222380544-c789468c-0603-43a1-871d-04773d85a5a7.png)

**Comment by LaurentPaquot** [comment]
I have the similar problem.
When I compile the first time my project the class is generated with internal object InjectedConfig which is fine (InjectedConfig is the name i have configured for the object)
I'm using kotlin 1.7.22, jdk 11, gradle 7.5.1 (I have enable the configuration-cache feature)
And when I'm running incremental builds, (sometimes) it re-generates the class with the 'actual' keyword. ("internal actual object InjectedConfig")
If I'm running my build with gradlew build --no-configuration-cache, it seems to regenerates the class properly.

Hope it will help you with the investigation for the potential bug fix.

**Comment by yshrsmz** [comment]
should be resolved with v0.14.0
https://github.com/yshrsmz/BuildKonfig/releases/tag/v0.14.0

**Comment by yshrsmz** [comment]
If you want a patch release for v0.13.x, feel free to send a PR.
